taichi/codegen/llvm/llvm_codegen_utils.cpp

```diff
@@ -31,7 +74,17 @@ void check_func_call_signature(llvm::Value *func,
   for (int i = 0; i < num_params; i++) {
     auto required = func_type->getFunctionParamType(i);
     auto provided = arglist[i]->getType();
+    /*
+     * Types in modules imported from files which are not the first appearances
+     * are renamed "original_type.xxx", so we have to create a pointer cast to
+     * the type in the function parameter list when the types in the function
+     * parameter are renamed.
```

Comment:
When importing a module from file, the imported llvm::Types can get conflict with the existing ones in the current llvm::Context. In such scenario, the imported types will get renamed from "original_type" to "original_type.xxx", making them separate types in essence. To recover the renamed types, a pointer cast to the original type was performed.